### ISSUE TITLE
Suspicious "Too many auth attempts"

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/mail#9156]
### Steps to reproduce

1. ???
2.
3.

### Expected behavior

Clean background syncs

### Actual behavior

`Too many auth attempts` in the logs for an account where the password hasn't changed and therefore should be correct.

### Mail app version

3.5 RC3

### PHP engine version

None

### Web server

None

### Database

None

**Comment by alexanderdd** [comment]
Probably the same issue? https://help.nextcloud.com/t/mail-sync-failed-mail-server-denied-authentication/154787

I think I also have this issue (or should I open a separate issue?):

NC 26.0.9, Mail 3.4.5 (I will update to 3.5.0 soon and report if it is still happening).

**Steps to reproduce**
* use mail app a while, open messages
* everything works fine
* at some point, message bodies don't get loaded anymore, instead the error message is shown "Nachricht konnte nicht geladen werden." (as I can see in https://github.com/nextcloud/mail/blob/main/l10n/de_DE.json this should be "Could not load the message" in English). Headers with sender and subject are still shown normally

I already contacted our Nextcloud hoster and our email provider, they say everything is fine on their side, must be a problem in Nextcloud Mail.

**Log**
```
[mail] Fehler: OCA\Mail\Exception\ServiceException: Sync failed for 38:INBOX: Too many auth attempts at <<closure>>

0. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 120
   OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 153
   OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
2. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
3. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
4. /var/www/html/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
5. /var/www/html/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["305","mail.mailboxes.sync"])
6. /var/www/html/lib/base.php line 1062
   OC\Route\Router->match("/apps/mail/api/mailboxes/305/sync")
7. /var/www/html/index.php line 36
   OC::handleRequest()

Caused by:

Horde_Imap_Client_Exception: Too many auth attempts at <<closure>>

 0. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
    OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 1613
    Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 4047
    Horde_Imap_Client_Base->status("*** sensitive parameters replaced ***", 8713)
 3. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 3647
    Horde_Imap_Client_Base->_syncStatus("*** sensitive parameters replaced ***")
 4. /var/www/html/custom_apps/mail/lib/IMAP/Sync/Synchronizer.php line 117
    Horde_Imap_Client_Base->sync("*** sensitive parameters replaced ***")
 5. /var/www/html/custom_apps/mail/lib/IMAP/Sync/Synchronizer.php line 74
    OCA\Mail\IMAP\Sync\Synchronizer->getNewMessageUids("*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 6. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 378
    OCA\Mail\IMAP\Sync\Synchronizer->sync("*** sensitive parameters replaced ***")
 7. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 244
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->runPartialSync("*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 8. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 120
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
 9. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 153
    OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
10. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
11. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
12. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
13. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["305","mail.mailboxes.sync"])
14. /var/www/html/lib/base.php line 1062
    OC\Route\Router->match("/apps/mail/api/mailboxes/305/sync")
15. /var/www/html/index.php line 36
    OC::handleRequest()

POST /apps/mail/api/mailboxes/305/sync
from 62.144.224.46 by alexander.mueller at 2023-12-19T14:48:55+01:00
``` 

I'm happy to provide more info if needed.

**Comment by ChristophWurst** [comment]
Possible explanation: mail servers limit the number of connections per IP. This we app may have many concurrent IMAP connections when the user causes many interactions with the UI. The server might reject IMAP connections when the limit is reached. Our app wrongly categorizes the error as authentication error. After a few failures, the app stops trying to connect.

@alexanderdd if you have something like a testing instance or a manageable single user instance you could turn on debug mode and analyze data/horde_imap.log entries written shortly before the first "Too many auth attempts" in data/nextcloud.log.

**Comment by ChristophWurst** [comment]
According to https://doc.dovecot.org/configuration_manual/authentication/ & https://doc.dovecot.org/configuration_manual/forwarding_parameters/ it might be possible to forward client IPs to make https://doc.dovecot.org/settings/core/#core_setting-mail_max_userip_connections work correctly.

**Comment by ChristophWurst** [comment]
Did a quick POC. This works by passing an `id` config array to the Horde constructor:

``C: 3 ID ("x-originating-ip" "1.2.3.4")``

Horde will identify if the `ID` capability is available.

**Comment by alexanderdd** [comment]
> @alexanderdd if you have something like a testing instance or a manageable single user instance you could turn on debug mode and analyze data/horde_imap.log entries written shortly before the first "Too many auth attempts" in data/nextcloud.log.

-> Sorry, I don't want to do this at the moment.



> Possible explanation: mail servers limit the number of connections per IP. This we app may have many concurrent IMAP connections when the user causes many interactions with the UI. The server might reject IMAP connections when the limit is reached. Our app wrongly categorizes the error as authentication error. After a few failures, the app stops trying to connect.

I also suspected that Nextcloud mail is triggering some kind of connection limits, but email provider says it doesn't, that's why I wrote "I already contacted our Nextcloud hoster and our email provider, they say everything is fine on their side, must be a problem in Nextcloud Mail." 

For more details, here is their answer, in German:
> Laut den Logs werden immer wieder innerhalb von einer Minute bis zu 17-19 Verbindungen zum Server geöffnet aber auch direkt geschlossen. Nach 11:43 Uhr gab es erst um 11:52:00 Uhr einen erneuten IMAP Login. 
> Wir haben noch mal mit den Kollegen aus der Fachabteilung gesprochen und den Fall entsprechend dargelegt. Auch die Kollegen können sich die Fehlermeldung nicht erklären. Es gibt zwar ein Limit, aber laut den Logs erreichen Sie das Limit von 30 Verbindungen pro Benutzer und pro IP nicht. Mehr Informationen können wir Ihnen bedauerlicherweise nicht zukommen lassen. 
> Darüber hinaus haben wir uns versichern lassen, dass es keine Änderungen an dem Mailsystem gegeben hat. Somit müssen wir von einem Clientseitigen Fehler ausgehen.

Auto-translated to English:
> According to the logs, up to 17-19 connections to the server are repeatedly opened within a minute but also closed immediately. After 11:43 a.m. there was only another IMAP login at 11:52:00 a.m.. 
> We spoke to our colleagues in the specialist department again and explained the case accordingly. They were also unable to explain the error message. There is a connection limit, but according to the logs you have not reached the limit of 30 connections per user and per IP. Unfortunately, we cannot provide you with any more information. 
> Furthermore, we have been assured that there have been no changes to the mail system. We must therefore assume that this is a client-side error.

**Comment by ferfebles** [comment]
Same problem here. In our nextcloud.log the problem appears while running background job:

"message":"Error while running background job (class: OCA\\Mail\\BackgroundJob\\QuotaJob, arguments: Array\n(\n    [accountId] => 361\n)\n)","userAgent":"--","version":"27.0.2.2","exception":{"Exception":"Horde_Imap_Client_Exception","Message":"Too many auth attempts"

I've tested to synchronize one of the accounts manually and I have the same error:

# sudo -u www-data /var/www/nextcloud/occ mail:account:sync 361 -vvv
[debug] Getting namespaces for account 361 failed: Too many auth attempts

In MailboxSync.php line 125:

  [OCA\Mail\Exception\ServiceException (102)]
  IMAP error synchronizing account 361: Too many auth attempts


Exception trace:
  at /var/www/nextcloud/apps/mail/lib/IMAP/MailboxSync.php:125
 OCA\Mail\IMAP\MailboxSync->sync() at /var/www/nextcloud/apps/mail/lib/Command/SyncAccount.php:103
 OCA\Mail\Command\SyncAccount->sync() at /var/www/nextcloud/apps/mail/lib/Command/SyncAccount.php:88
 OCA\Mail\Command\SyncAccount->execute() at /var/www/nextcloud/3rdparty/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /var/www/nextcloud/lib/private/Console/Application.php:211
 OC\Console\Application->run() at /var/www/nextcloud/console.php:100
 require_once() at /var/www/nextcloud/occ:11

In ImapClientRateLimitingDecorator.php line 66:

  [Horde_Imap_Client_Exception (102)]
  Too many auth attempts


Exception trace:
  at /var/www/nextcloud/apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php:66
 OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login() at /var/www/nextcloud/apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php:853
 Horde_Imap_Client_Base->login() at /var/www/nextcloud/apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php:1404
 Horde_Imap_Client_Base->listMailboxes() at /var/www/nextcloud/apps/mail/lib/IMAP/FolderMapper.php:59
 OCA\Mail\IMAP\FolderMapper->getFolders() at /var/www/nextcloud/apps/mail/lib/IMAP/MailboxSync.php:122
 OCA\Mail\IMAP\MailboxSync->sync() at /var/www/nextcloud/apps/mail/lib/Command/SyncAccount.php:103
 OCA\Mail\Command\SyncAccount->sync() at /var/www/nextcloud/apps/mail/lib/Command/SyncAccount.php:88
 OCA\Mail\Command\SyncAccount->execute() at /var/www/nextcloud/3rdparty/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /var/www/nextcloud/lib/private/Console/Application.php:211
 OC\Console\Application->run() at /var/www/nextcloud/console.php:100
 require_once() at /var/www/nextcloud/occ:11

mail:account:sync [-f|--force] [--] <account-id>

**Comment by ChristophWurst** [comment]
> "Too many auth attempts"

The problem occurs earlier than that. First the app fails to authenticate three times. Then the internal rate limiting kicks in and no longer attempts to try an IMAP login.

Scan your log file for Mail/Horde errors before "Too many auth attempts".

**Comment by drfinkus** [comment]
I believe I am experiencing the same problem with Mail 3.5.0.

I see the following errors in the console:
```
MessageService.js:71 
        
        
       POST https://cloud.fwdeu.org/apps/mail/api/mailboxes/17/sync 500 (Internal Server Error)
(anonymous) @ xhr.js:251
[...]
actions.js:842 Could not sync envelopes: Request failed with status code 500 a {message: 'Request failed with status code 500', name: 'AxiosError', code: 'ERR_BAD_RESPONSE', config: {…}, request: XMLHttpRequest, …}
[...]
App.vue:86 [ERROR] mail: Background sync failed: Request failed with status code 500 {app: 'mail', uid: 'xxxxx', level: '2', error: a}
```

In the Logging section, I see _many_ of these errors, which started to occur around the same time when the 

```
[mail] Error: Horde_Imap_Client_Exception: Too many auth attempts at <<closure>>

0. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
   OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 222
   Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
2. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 129
   OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
3. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 156
   OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
4. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
5. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
6. /var/www/html/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
7. /var/www/html/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["17","mail.mailboxes.sync"])
8. /var/www/html/lib/base.php line 1068
   OC\Route\Router->match("/apps/mail/api/mailboxes/17/sync")
9. /var/www/html/index.php line 38
   OC::handleRequest()

POST /apps/mail/api/mailboxes/17/sync
from 193.203.xxx.xxx by xxxxxxx at 2024-01-06T19:29:29+00:00
```

I can confirm what @ChristophWurst mentions above. Right before these errors have started, I see three of the following errors:
```
[mail] Error: OCA\Mail\Exception\ServiceException: Could not get message flag options from IMAP: Mail server denied authentication. at <<closure>>

 0. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 795
    OCA\Mail\Service\MailManager->isPermflagsEnabled(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "INBOX")
 1. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 470
    OCA\Mail\Service\MailManager->filterFlags(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "$label1", "INBOX")
 2. /var/www/html/custom_apps/mail/lib/Listener/NewMessageClassificationListener.php line 127
    OCA\Mail\Service\MailManager->flagMessage("*** sensitive parameters replaced ***", "INBOX", 1132, "$label1", "*** sensitive parameters replaced ***")
 3. /var/www/html/lib/private/EventDispatcher/ServiceEventListener.php line 86
    OCA\Mail\Listener\NewMessageClassificationListener->handle(["OCA\\Mail\\Eve ... "])
 4. /var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 251
    OC\EventDispatcher\ServiceEventListener->__invoke(["OCA\\Mail\\Eve ... "], "OCA\\Mail\\Even ... d", ["Symfony\\Compo ... "])
 5. /var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 73
    Symfony\Component\EventDispatcher\EventDispatcher->callListeners([["Closure"],["Closure"]], "OCA\\Mail\\Even ... d", ["OCA\\Mail\\Eve ... "])
 6. /var/www/html/lib/private/EventDispatcher/EventDispatcher.php line 94
    Symfony\Component\EventDispatcher\EventDispatcher->dispatch(["OCA\\Mail\\Eve ... "], "OCA\\Mail\\Even ... d")
 7. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 432
    OC\EventDispatcher\EventDispatcher->dispatch("OCA\\Mail\\Even ... d", ["OCA\\Mail\\Eve ... "])
 8. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 261
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->runPartialSync(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 9. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 129
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
10. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 156
    OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
11. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
12. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
13. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
14. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["17","mail.mailboxes.sync"])
15. /var/www/html/lib/base.php line 1068
    OC\Route\Router->match("/apps/mail/api/mailboxes/17/sync")
16. /var/www/html/index.php line 38
    OC::handleRequest()

Caused by:

Horde_Imap_Client_Exception: Mail server denied authentication. at <<closure>>

 0. /var/www/html/custom_apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php line 73
    Horde_Imap_Client_Socket->_login("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
    OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 1613
    Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
 3. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 810
    Horde_Imap_Client_Base->status("INBOX", 256)
 4. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 795
    OCA\Mail\Service\MailManager->isPermflagsEnabled(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "INBOX")
 5. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 470
    OCA\Mail\Service\MailManager->filterFlags(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "$label1", "INBOX")
 6. /var/www/html/custom_apps/mail/lib/Listener/NewMessageClassificationListener.php line 127
    OCA\Mail\Service\MailManager->flagMessage("*** sensitive parameters replaced ***", "INBOX", 1132, "$label1", "*** sensitive parameters replaced ***")
 7. /var/www/html/lib/private/EventDispatcher/ServiceEventListener.php line 86
    OCA\Mail\Listener\NewMessageClassificationListener->handle(["OCA\\Mail\\Eve ... "])
 8. /var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 251
    OC\EventDispatcher\ServiceEventListener->__invoke(["OCA\\Mail\\Eve ... "], "OCA\\Mail\\Even ... d", ["Symfony\\Compo ... "])
 9. /var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php line 73
    Symfony\Component\EventDispatcher\EventDispatcher->callListeners([["Closure"],["Closure"]], "OCA\\Mail\\Even ... d", ["OCA\\Mail\\Eve ... "])
10. /var/www/html/lib/private/EventDispatcher/EventDispatcher.php line 94
    Symfony\Component\EventDispatcher\EventDispatcher->dispatch(["OCA\\Mail\\Eve ... "], "OCA\\Mail\\Even ... d")
11. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 432
    OC\EventDispatcher\EventDispatcher->dispatch("OCA\\Mail\\Even ... d", ["OCA\\Mail\\Eve ... "])
12. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 261
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->runPartialSync(["OCA\\Mail\\IMA ... "], "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
13. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 129
    OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
14. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 156
    OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
15. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
16. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
17. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
18. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["17","mail.mailboxes.sync"])
19. /var/www/html/lib/base.php line 1068
    OC\Route\Router->match("/apps/mail/api/mailboxes/17/sync")
20. /var/www/html/index.php line 38
    OC::handleRequest()

POST /apps/mail/api/mailboxes/17/sync
from 193.203.xxx.xxx by xxxxxxx at 2024-01-06T18:42:15+00:00
```

Now, I'm not sure what happened, the server works fine, the credentials are unchanged, and the mailbox works fine on mobile and other clients.

To help debug, I installed and activated Snappymail in parallel, and logged in to the same mailbox which Mail is unable to log into. Snappymail works fine, which leads me to believe that the problem is on the Mail side, not on the server side.

Bottom line, I have three questions:
1. If the app no longer attempts to login after three failed attempts, is there a way to reset the counter back? Or does it time out by itself and starts reattempting to authenticate after a while?
2. Can the internal rate limiting be configured or removed altogether?
3. What could have caused the failed authentication, when the mailbox credentials have not changed and Snappymail is able to access it just fine?

**Comment by ChristophWurst** [comment]
> To help debug, I installed and activated Snappymail in parallel, and logged in to the same mailbox which Mail is unable to log into. Snappymail works fine, which leads me to believe that the problem is on the Mail side, not on the server side.

Does Mail and Snappymail connect from the same client IP? If the IMAP server blocks the Mail IP due to too many logins/connections it should blocky Snappymail too.

> 1. If the app no longer attempts to login after three failed attempts, is there a way to reset the counter back? Or does it time out by itself and starts reattempting to authenticate after a while?

It's a 3h sliding window. The limit times out.

> Can the internal rate limiting be configured or removed altogether?

You can't unless you patch `\OCA\Mail\IMAP\ImapClientRateLimitingDecorator::_login` manually.

> What could have caused the failed authentication, when the mailbox credentials have not changed and Snappymail is able to access it just fine?

See the ticket history. It's possibly caused by too many connections from one IP.

**Comment by drfinkus** [comment]
> Does Mail and Snappymail connect from the same client IP? If the IMAP server blocks the Mail IP due to too many logins/connections it should blocky Snappymail too.

Yes, Mail and Snappymail connect from the same client IP. I have them side by side, right now, same client IP, same server, Snappymail works fine, Mail doesn't, same error, too many auth attempts. It does not seem like the IMAP server is blocking the Mail IP, otherwise Snappymail should not work either.

What is also strange is that it seems to work for a while, and then it stops working. I was able to use it today to send a couple of emails, but now it stopped working again. Basically when the window resets, I can use it briefly, then it breaks again. Snappymail does not face any issue, again, same IP.

The current period of "too many auth attempts" started 10 minutes ago with the following error:
```
[mail] Error: OCA\Mail\Exception\ServiceException: Could not fetch stats of mailbox 17. IMAP error: Mail server denied authentication. at <<closure>>

0. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 132
   OCA\Mail\IMAP\MailboxSync->syncStats("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 156
   OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
2. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
3. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
4. /var/www/html/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
5. /var/www/html/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["17","mail.mailboxes.sync"])
6. /var/www/html/lib/base.php line 1068
   OC\Route\Router->match("/apps/mail/api/mailboxes/17/sync")
7. /var/www/html/index.php line 38
   OC::handleRequest()

Caused by:

Horde_Imap_Client_Exception: Mail server denied authentication. at <<closure>>

 0. /var/www/html/custom_apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php line 73
    Horde_Imap_Client_Socket->_login("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
    OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 1613
    Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
 3. /var/www/html/custom_apps/mail/lib/IMAP/FolderMapper.php line 141
    Horde_Imap_Client_Base->status(["INBOX"])
 4. /var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php line 166
    OCA\Mail\IMAP\FolderMapper->getFoldersStatusAsObject(["OCA\\Mail\\IMA ... "], ["INBOX"])
 5. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 132
    OCA\Mail\IMAP\MailboxSync->syncStats("*** sensitive parameters replaced ***")
 6. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 156
    OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
 7. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
 8. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
 9. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
10. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["17","mail.mailboxes.sync"])
11. /var/www/html/lib/base.php line 1068
    OC\Route\Router->match("/apps/mail/api/mailboxes/17/sync")
12. /var/www/html/index.php line 38
    OC::handleRequest()

POST /apps/mail/api/mailboxes/17/sync
from 194.5.xx.xx by xxxxxx at 2024-01-08T15:23:08+00:00
```

**Comment by ChristophWurst** [comment]
> Snappymail works fine, Mail doesn't, same error, too many auth attempts

You have to try Snappymail right when Mail has the initial three auth errors, not when it switches over to "Too many auth attempts". At the latter, there are no more IMAP connections. The app bails early.

**Comment by ChristophWurst** [comment]
If you are the only user on that Nextcloud installation please enable debug mode and wait for the error to happen again after the window reset. Then inspect the data/horde_imap.log and look for the entries around the three timestamps where authentication failed, or the last entry before "Too many auth attempts" in data/nextcloud.log. It would be super helpful to see the IMAP communication and hopefully more details about the error.

**Comment by ChristophWurst** [comment]
Two attempts to trigger the failing auth locally
1) Run the `occ mail:account:diagnose` command in an endless loop in four terminal sessions
2) Duplicate the oc_jobs quota job entry of one account hundres of time so that there is one cron.php run with many quota jobs

In both cases I was not able to have failing auth. The same IMAP account occasionally fails to authenticate on a production Nextcloud installation.

**Comment by ChristophWurst** [comment]
I am realizing that Horde throws a LOGIN_AUTHENTICATIONFAILED when the authentication fails, but also when it doesn't receive a login error. Then it defaults to LOGIN_AUTHENTICATIONFAILED too. For the rate limiting we should only consider the former.

**Comment by ChristophWurst** [comment]
@alexanderdd @drfinkus I am unable to reproduce the error myself so I will rely on your testing: https://github.com/nextcloud/mail/pull/9232. https://docs.nextcloud.com/server/latest/admin_manual/issues/applying_patch.html#patching-apps for guidance.

**Comment by drfinkus** [comment]
Thanks @ChristophWurst, I applied the patch now, let's see how it goes, I'll let you know tomorrow.

**Comment by drfinkus** [comment]
Seems to work well, I haven't encountered any problems since the patch.

**Comment by alexanderdd** [comment]
Thanks for working on this!

I cannot apply the patch manually, I will wait for a new release of the app that includes the merged PR, and report back if it fixes our issue.

currently using NC 27.1.5 and mail 3.5.2

I still have this issue. Here are three examples of the errors:

```
[core] Fehler: Horde_Imap_Client_Exception: Too many auth attempts at <<closure>>

0. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
   OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 1404
   Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
2. /var/www/html/custom_apps/mail/lib/IMAP/FolderMapper.php line 58
   Horde_Imap_Client_Base->listMailboxes("*", 5, [true,true,true])
3. /var/www/html/custom_apps/mail/lib/Service/MailManager.php line 598
   OCA\Mail\IMAP\FolderMapper->getFolders(["OCA\\Mail\\Account"], ["OCA\\Mail\\IMA ... e])
4. /var/www/html/custom_apps/mail/lib/BackgroundJob/QuotaJob.php line 92
   OCA\Mail\Service\MailManager->getQuota(["OCA\\Mail\\Account"])
5. /var/www/html/lib/public/BackgroundJob/Job.php line 81
   OCA\Mail\BackgroundJob\QuotaJob->run([61])
6. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start(["OC\\BackgroundJob\\JobList"])
7. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start(["OC\\BackgroundJob\\JobList"])
8. /var/www/html/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute(["OC\\BackgroundJob\\JobList"], ["OC\\Log"])

at 2024-01-11T20:41:17+01:00



[mail] Fehler: Horde_Imap_Client_Exception: Mail server denied authentication. at <<closure>>

0. /var/www/html/custom_apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php line 73
   Horde_Imap_Client_Socket->_login("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
   OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
2. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 222
   Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
3. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 137
   OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
4. /var/www/html/custom_apps/mail/lib/BackgroundJob/SyncJob.php line 111
   OCA\Mail\Service\Sync\ImapToDbSynchronizer->syncAccount("*** sensitive parameters replaced ***")
5. /var/www/html/lib/public/BackgroundJob/Job.php line 81
   OCA\Mail\BackgroundJob\SyncJob->run([39])
6. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 103
   OCP\BackgroundJob\Job->start(["OC\\BackgroundJob\\JobList"])
7. /var/www/html/lib/public/BackgroundJob/TimedJob.php line 93
   OCP\BackgroundJob\TimedJob->start(["OC\\BackgroundJob\\JobList"])
8. /var/www/html/cron.php line 152
   OCP\BackgroundJob\TimedJob->execute(["OC\\BackgroundJob\\JobList"], ["OC\\Log"])

at 2024-01-11T19:14:28+01:00






[mail] Fehler: Horde_Imap_Client_Exception: Too many auth attempts at <<closure>>

0. /var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php line 853
   OCA\Mail\IMAP\ImapClientRateLimitingDecorator->_login("*** sensitive parameters replaced ***")
1. /var/www/html/custom_apps/mail/lib/Service/Sync/ImapToDbSynchronizer.php line 222
   Horde_Imap_Client_Base->login("*** sensitive parameters replaced ***")
2. /var/www/html/custom_apps/mail/lib/Service/Sync/SyncService.php line 123
   OCA\Mail\Service\Sync\ImapToDbSynchronizer->sync("*** sensitive parameters replaced ***")
3. /var/www/html/custom_apps/mail/lib/Controller/MailboxesController.php line 155
   OCA\Mail\Service\Sync\SyncService->syncMailbox("*** sensitive parameters replaced ***")
4. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
   OCA\Mail\Controller\MailboxesController->sync("*** sensitive parameters replaced ***")
5. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
   OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Mail\\Con ... "], "sync")
6. /var/www/html/lib/private/AppFramework/App.php line 183
   OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Mail\\Con ... "], "sync")
7. /var/www/html/lib/private/Route/Router.php line 315
   OC\AppFramework\App::main("OCA\\Mail\\Cont ... r", "sync", ["OC\\AppFramewo ... "], ["128","mail.mailboxes.sync"])
8. /var/www/html/lib/base.php line 1068
   OC\Route\Router->match("/apps/mail/api/mailboxes/128/sync")
9. /var/www/html/index.php line 38
   OC::handleRequest()

POST /apps/mail/api/mailboxes/128/sync
from 212.172.2.141 by *** sensitive parameters replaced *** at 2024-01-11T15:48:39+01:00
```

**Comment by ChristophWurst** [comment]
> I cannot apply the patch manually

Does it not apply?

**Comment by drfinkus** [comment]
> But I'm assuming you still have the occasional "Mail server denied authentication." in your log, right?

Yes, that's true. And now the issue came back, but from another mailbox. There's approx. 20 mailboxes on the server. Maybe the reason why it seems to have slowed down is various mailboxes are triggering the same error but it is rarer statistically that they happen within a short interval of each other?

I will look into enabling debugging, I'm a bit unclear how to do that.

**Comment by drfinkus** [comment]
Please disregard my previous message. What actually happened was that the Mail app got automatically upgraded to 3.5.2 (AIO), which of course reverted `ImapClientRateLimitingDecorator.php` back to its pre-patch version. 

I reapplied the patch now, all seems well. But yes, to answer your question, @ChristophWurst, I do see the occasional `Mail server denied authentication`, I'll turn on debug to see if I can get more details.

**Comment by drfinkus** [comment]
Here is one example of what I'm seeing in the logs.

```
{
    "reqId": "mKOdseJ13PgpIMNxgyEX",
    "level": 1,
    "time": "2024-01-14T13:43:20+00:00",
    "remoteAddr": "",
    "user": "--",
    "app": "mail",
    "method": "",
    "url": "--",
    "message": "Cron mail sync authentication failed for account 18",
    "userAgent": "--",
    "version": "27.1.5.1",
    "exception": {
        "Exception": "OCA\\Mail\\Exception\\ServiceException",
        "Message": "IMAP error synchronizing account 18: Mail server denied authentication.",
        "Code": 102,
        "Trace": [
            {
                "file": "/var/www/html/custom_apps/mail/lib/BackgroundJob/SyncJob.php",
                "line": 110,
                "function": "sync",
                "class": "OCA\\Mail\\IMAP\\MailboxSync",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/Job.php",
                "line": 81,
                "function": "run",
                "class": "OCA\\Mail\\BackgroundJob\\SyncJob",
                "type": "->",
                "args": [
                    [
                        18
                    ]
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                "line": 103,
                "function": "start",
                "class": "OCP\\BackgroundJob\\Job",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ]
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                "line": 93,
                "function": "start",
                "class": "OCP\\BackgroundJob\\TimedJob",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ]
                ]
            },
            {
                "file": "/var/www/html/cron.php",
                "line": 152,
                "function": "execute",
                "class": "OCP\\BackgroundJob\\TimedJob",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ],
                    [
                        "OC\\Log"
                    ]
                ]
            }
        ],
        "File": "/var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php",
        "Line": 125,
        "Previous": {
            "Exception": "Horde_Imap_Client_Exception",
            "Message": "Mail server denied authentication.",
            "Code": 102,
            "Trace": [
                {
                    "file": "/var/www/html/custom_apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php",
                    "line": 73,
                    "function": "_login",
                    "class": "Horde_Imap_Client_Socket",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
                    "line": 853,
                    "function": "_login",
                    "class": "OCA\\Mail\\IMAP\\ImapClientRateLimitingDecorator",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
                    "line": 1404,
                    "function": "login",
                    "class": "Horde_Imap_Client_Base",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/var/www/html/custom_apps/mail/lib/IMAP/FolderMapper.php",
                    "line": 59,
                    "function": "listMailboxes",
                    "class": "Horde_Imap_Client_Base",
                    "type": "->",
                    "args": [
                        "*",
                        5,
                        [
                            "*** sensitive parameters replaced ***",
                            "*** sensitive parameters replaced ***",
                            "*** sensitive parameters replaced ***"
                        ]
                    ]
                },
                {
                    "file": "/var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php",
                    "line": 122,
                    "function": "getFolders",
                    "class": "OCA\\Mail\\IMAP\\FolderMapper",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***",
                        [
                            "OCA\\Mail\\IMAP\\ImapClientRateLimitingDecorator",
                            [
                                "HICenv",
                                "HICflags",
                                "HIChdrs",
                                "HICdate",
                                "HICsize",
                                "HICstruct"
                            ],
                            "*** sensitive parameters replaced ***",
                            "*** sensitive parameters replaced ***"
                        ]
                    ]
                },
                {
                    "file": "/var/www/html/custom_apps/mail/lib/BackgroundJob/SyncJob.php",
                    "line": 110,
                    "function": "sync",
                    "class": "OCA\\Mail\\IMAP\\MailboxSync",
                    "type": "->",
                    "args": [
                        "*** sensitive parameters replaced ***"
                    ]
                },
                {
                    "file": "/var/www/html/lib/public/BackgroundJob/Job.php",
                    "line": 81,
                    "function": "run",
                    "class": "OCA\\Mail\\BackgroundJob\\SyncJob",
                    "type": "->",
                    "args": [
                        [
                            18
                        ]
                    ]
                },
                {
                    "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                    "line": 103,
                    "function": "start",
                    "class": "OCP\\BackgroundJob\\Job",
                    "type": "->",
                    "args": [
                        [
                            "OC\\BackgroundJob\\JobList"
                        ]
                    ]
                },
                {
                    "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                    "line": 93,
                    "function": "start",
                    "class": "OCP\\BackgroundJob\\TimedJob",
                    "type": "->",
                    "args": [
                        [
                            "OC\\BackgroundJob\\JobList"
                        ]
                    ]
                },
                {
                    "file": "/var/www/html/cron.php",
                    "line": 152,
                    "function": "execute",
                    "class": "OCP\\BackgroundJob\\TimedJob",
                    "type": "->",
                    "args": [
                        [
                            "OC\\BackgroundJob\\JobList"
                        ],
                        [
                            "OC\\Log"
                        ]
                    ]
                }
            ],
            "File": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
            "Line": 572
        },
        "message": "Cron mail sync authentication failed for account {accountId}",
        "exception": {},
        "CustomMessage": "Cron mail sync authentication failed for account 18"
    }
}
{
    "reqId": "wYt6NJzjq44pfoSjnDk4",
    "level": 0,
    "time": "2024-01-14T13:48:14+00:00",
    "remoteAddr": "",
    "user": "--",
    "app": "mail",
    "method": "",
    "url": "--",
    "message": "Getting namespaces for account 21 failed: Authentication failed.",
    "userAgent": "--",
    "version": "27.1.5.1",
    "exception": {
        "Exception": "Horde_Imap_Client_Exception",
        "Message": "Authentication failed.",
        "Code": 102,
        "Trace": [
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
                "line": 4584,
                "function": "_responseCode",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    [
                        "Horde_Imap_Client_Interaction_Pipeline",
                        [
                            [],
                            [],
                            [
                                "Horde_Imap_Client_Exception",
                                null,
                                false,
                                "Authentication failed."
                            ]
                        ],
                        [
                            "Horde_Imap_Client_Fetch_Results"
                        ]
                    ],
                    [
                        "Horde_Imap_Client_Interaction_Server_Tagged",
                        [
                            "stdClass",
                            [],
                            "AUTHENTICATIONFAILED"
                        ],
                        3,
                        [
                            "Horde_Imap_Client_Tokenize"
                        ],
                        "1"
                    ]
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
                "line": 4319,
                "function": "_getLine",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    [
                        "Horde_Imap_Client_Interaction_Pipeline",
                        [
                            [],
                            [],
                            [
                                "Horde_Imap_Client_Exception",
                                null,
                                false,
                                "Authentication failed."
                            ]
                        ],
                        [
                            "Horde_Imap_Client_Fetch_Results"
                        ]
                    ]
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
                "line": 4243,
                "function": "_sendCmdChunk",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    [
                        "Horde_Imap_Client_Interaction_Pipeline",
                        [
                            [],
                            [],
                            [
                                "Horde_Imap_Client_Exception",
                                null,
                                false,
                                "Authentication failed."
                            ]
                        ],
                        [
                            "Horde_Imap_Client_Fetch_Results"
                        ]
                    ],
                    [
                        [
                            "Horde_Imap_Client_Interaction_Command",
                            [],
                            "*** sensitive parameters replaced ***",
                            false,
                            null,
                            null,
                            [
                                "Horde_Imap_Client_Interaction_Pipeline",
                                [
                                    [],
                                    [],
                                    [
                                        "Horde_Imap_Client_Exception",
                                        null,
                                        false,
                                        "Authentication failed."
                                    ]
                                ],
                                [
                                    "Horde_Imap_Client_Fetch_Results"
                                ]
                            ],
                            [
                                "Horde_Imap_Client_Interaction_Server_Tagged",
                                [
                                    "stdClass",
                                    [],
                                    "AUTHENTICATIONFAILED"
                                ],
                                3,
                                [
                                    "Horde_Imap_Client_Tokenize"
                                ],
                                "1"
                            ],
                            "1"
                        ]
                    ]
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
                "line": 851,
                "function": "_sendCmd",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    [
                        "Horde_Imap_Client_Interaction_Pipeline",
                        [
                            [],
                            [],
                            [
                                "Horde_Imap_Client_Exception",
                                null,
                                false,
                                "Authentication failed."
                            ]
                        ],
                        [
                            "Horde_Imap_Client_Fetch_Results"
                        ]
                    ]
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
                "line": 512,
                "function": "_tryLogin",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/lib/IMAP/ImapClientRateLimitingDecorator.php",
                "line": 73,
                "function": "_login",
                "class": "Horde_Imap_Client_Socket",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
                "line": 853,
                "function": "_login",
                "class": "OCA\\Mail\\IMAP\\ImapClientRateLimitingDecorator",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
                "line": 750,
                "function": "login",
                "class": "Horde_Imap_Client_Base",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/lib/IMAP/MailboxSync.php",
                "line": 107,
                "function": "getNamespaces",
                "class": "Horde_Imap_Client_Base",
                "type": "->",
                "args": [
                    [],
                    [
                        "*** sensitive parameters replaced ***"
                    ]
                ]
            },
            {
                "file": "/var/www/html/custom_apps/mail/lib/BackgroundJob/SyncJob.php",
                "line": 110,
                "function": "sync",
                "class": "OCA\\Mail\\IMAP\\MailboxSync",
                "type": "->",
                "args": [
                    "*** sensitive parameters replaced ***"
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/Job.php",
                "line": 81,
                "function": "run",
                "class": "OCA\\Mail\\BackgroundJob\\SyncJob",
                "type": "->",
                "args": [
                    [
                        21
                    ]
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                "line": 103,
                "function": "start",
                "class": "OCP\\BackgroundJob\\Job",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ]
                ]
            },
            {
                "file": "/var/www/html/lib/public/BackgroundJob/TimedJob.php",
                "line": 93,
                "function": "start",
                "class": "OCP\\BackgroundJob\\TimedJob",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ]
                ]
            },
            {
                "file": "/var/www/html/cron.php",
                "line": 152,
                "function": "execute",
                "class": "OCP\\BackgroundJob\\TimedJob",
                "type": "->",
                "args": [
                    [
                        "OC\\BackgroundJob\\JobList"
                    ],
                    [
                        "OC\\Log"
                    ]
                ]
            }
        ],
        "File": "/var/www/html/custom_apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Socket.php",
        "Line": 5048,
        "message": "Getting namespaces for account 21 failed: Authentication failed.",
        "exception": {
            "details": null,
            "logged": false,
            "raw_msg": "Authentication failed."
        },
        "CustomMessage": "Getting namespaces for account 21 failed: Authentication failed."
    }
}
```

**Comment by drfinkus** [comment]
The plot thickens. This is what the mail provider sees in their logs at the same timestamp as this error: https://github.com/nextcloud/mail/issues/9156#issuecomment-1890985870

```
Jan 14 12:23:06 imap2 dovecot: imap(my_inbox_username)<7802><wHBc9eYOZOpOn3DD>: Disconnected: Connection closed (No commands sent) in=0 out=416 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0
```

**Comment by drfinkus** [comment]
@ChristophWurst I can confirm that #9232 fixes the issue for me. It would be great if we could get that patch merged, because now I have to manually reapply it every time the mail app gets upgraded, which it does quite frequently :)

**Comment by Fantu** [comment]
Hi, I have a nextcloud 29.0.16 little used and with few users with mail app 3.7.24 and with hundreds of logs per day of which over 95% are about this warning and all in groups of 4-6 entries within 1-2 seconds.

I understand from a quick search that a limitation had been applied, but I have the doubt that it doesn't work, I post one of those complete entries:

```
{
  "reqId": "zrLf8VNvMwHKIxk73NhD",
  "level": 2,
  "time": "2025-04-29T11:15:03+00:00",
  "remoteAddr": "",
  "user": "--",
  "app": "mail",
  "method": "",
  "url": "--",
  "message": "Could not fetch structure detail data to enhance message previews: Too many auth attempts",
  "userAgent": "--",
  "version": "29.0.16.1",
  "exception": {
    "Exception": "Horde_Imap_Client_Exception",
    "Message": "Too many auth attempts",
    "Code": 102,
    "Trace": [
      {
        "file": "/var/www/nextcloud/apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
        "line": 853,
        "function": "_login",
        "class": "OCA\\Mail\\IMAP\\HordeImapClient",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
        "line": 2510,
        "function": "login",
        "class": "Horde_Imap_Client_Base",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/nextcloud/apps/mail/vendor/bytestream/horde-imap-client/lib/Horde/Imap/Client/Base.php",
        "line": 2493,
        "function": "_fetchWrapper",
        "class": "Horde_Imap_Client_Base",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/mail/lib/IMAP/MessageMapper.php",
        "line": 889,
        "function": "fetch",
        "class": "Horde_Imap_Client_Base",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/mail/lib/IMAP/PreviewEnhancer.php",
        "line": 86,
        "function": "getBodyStructureData",
        "class": "OCA\\Mail\\IMAP\\MessageMapper",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/mail/lib/Service/PreprocessingService.php",
        "line": 83,
        "function": "process",
        "class": "OCA\\Mail\\IMAP\\PreviewEnhancer",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/apps/mail/lib/BackgroundJob/PreviewEnhancementProcessingJob.php",
        "line": 93,
        "function": "process",
        "class": "OCA\\Mail\\Service\\PreprocessingService",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/Job.php",
        "line": 80,
        "function": "run",
        "class": "OCA\\Mail\\BackgroundJob\\PreviewEnhancementProcessingJob",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php",
        "line": 107,
        "function": "start",
        "class": "OCP\\BackgroundJob\\Job",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php",
        "line": 94,
        "function": "start",
        "class": "OCP\\BackgroundJob\\TimedJob",
        "type": "->"
      },
      {
        "file": "/var/www/nextcloud/cron.php",
        "line": 177,
        "function": "execute",
        "class": "OCP\\BackgroundJob\\TimedJob",
        "type": "->"
      }
    ],
    "File": "/var/www/nextcloud/apps/mail/lib/IMAP/HordeImapClient.php",
    "Line": 69,
    "message": "Could not fetch structure detail data to enhance message previews: Too many auth attempts",
    "exception": {
      "details": null,
      "logged": false,
      "raw_msg": "Too many auth attempts"
    },
    "CustomMessage": "Could not fetch structure detail data to enhance message previews: Too many auth attempts"
  },
  "id": "6810bb2e9b3ea"
}
```

**Comment by ChristophWurst** [comment]
Thanks for the report, @Fantu. As this ticket is more than a year old, please create a new report with your information. Thank you.

**Comment by ChristophWurst** [comment]
And you are running Nextcloud 29 which has reached end of life. Please update to 30.

**Comment by ChristophWurst** [comment]
Makes sense. Yes please create a new ticket.
